### PR TITLE
Surface subzone observed species data, and build rollup data with them

### PR DESCRIPTION
**Pull request by tommylau523**

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2720** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2720?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2719** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2719?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2719?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#2716** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2716?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by sgrimm** [comment]
This new logic isn't covered by `ObservationResultsStoreTest`; is that intentional?

**Comment by tommylau523** [comment]
I am still working on the spreadsheet to add the data. I should probably have put this in Draft.

**Comment by tommylau523** [comment]
Finally done! A lot of calculations so it took a while

**Comment by sgrimm** [comment]
@alexTerraformation , do you want to review the spreadsheets here to make sure the updates match your intent from the requirements? I'll look at them either way but I may miss problems you'd notice.

**Comment by alexTerraformation** [comment]
> @alexTerraformation , do you want to review the spreadsheets here to make sure the updates match your intent from the requirements? I'll look at them either way but I may miss problems you'd notice.

Sure thing, I will review this afternoon!

File: src/test/kotlin/com/terraformation/backend/tracking/db/ObservationResultsStoreTest.kt [hunk 1]
```diff
@@ -429,6 +421,71 @@ class ObservationResultsStoreTest : DatabaseTest(), RunsAsUser {
...
+                    .flatMap { subzone -> subzone.monitoringPlots }
+                    .firstOrNull { it.monitoringPlotNumber == plotNumber.toLong() }
+
+            listOf(
+                plot?.totalPlants.toStringOrBlank(),
+                plot?.totalSpecies.toStringOrBlank(),
+                plot?.mortalityRate.toStringOrBlank("%"),
+                // Live and existing plants columns are in spreadsheet but not included in
+                // calculated
+                // results; it will be removed by the filter function below.
```
Comment (sgrimm) [inline src/test/kotlin/com/terraformation/backend/tracking/db/ObservationResultsStoreTest.kt]: Nit: Could get rid of the line break here.

**Review by sgrimm — APPROVED** [review]
This looks good! Alex is still going to sanity check the spreadsheets, but the code seems right to me.

**Review by alexTerraformation — APPROVED** [review]
I reviewed the source google sheet and it looked good to me.  Thanks @tommylau523 !

**Comment by tommylau523** [comment]
### Merge activity

* **Dec 30, 11:04 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2719).
* **Dec 30, 11:05 PM EST**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2719).

**Pull request merged**